lib/board.rb

```diff
@@ -0,0 +1,60 @@
+class Board 
+  attr_accessor :cells 
+  
+  
+  def initialize
+    @cells = Array.new(9," ")
+  end
+
+  def reset!
+    self.cells =[" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+    ## line spacing
+    puts ""
+    puts""
+  end
+
+
+  def position(position)
+     @cells[position.to_i-1]
+  end
+
+  def full?
+    @cells.all?{|piece|piece == "X" || piece == "O"}
+  end
+
+  def turn_count
+    x=0
+    @cells.each do |piece|
+      if piece == "X" || piece == "O"
+        x+=1
+      end
+    end
+    x
+  end
+
+  def taken?(position)
+    if @cells[position.to_i-1] == "X" || @cells[position.to_i-1] == "O"
+      true
+  else
+    false
```

Comment:
Avoid returning boolean literals from `if` statements, relying instead on the implicit return value of the boolean expression.
```
def taken?(position)
  @cells[position.to_i-1] == "X" || @cells[position.to_i-1] == "O"
end
```